### PR TITLE
Feat(agent label): use a specific label for windows if not a container

### PR DESCRIPTION
**Pull request by smerle33**
in order to avoid the `windows` label as per https://github.com/jenkins-infra/helpdesk/issues/3223#issuecomment-1305703973 we set a specific label.

### CODE REVIEW

File: vars/buildPlugin.groovy [hunk 1]
```diff
@@ -29,24 +29,31 @@ def call(Map params = [:]) {
...
+      if (platform == 'linux' || platform == 'windows') {
+        def agentContainerLabel = jdk == '8' ? 'maven' : 'maven-' + jdk
+        if (platform == 'windows') {
+          agentContainerLabel += '-windows'
+        }
+        label = agentContainerLabel
+      }
+    } else {
+      if (platform == 'windows') {
+        label = 'docker-windows'
```
[the remark this answers is not in the document]
Comment (smerle33) [reply vars/buildPlugin.groovy:55]: as discussed privately this should not be the solution as we aim to avoid the label "windows" ...

**Comment by smerle33** [comment]
checked here https://github.com/jenkinsci/jenkins-infra-test-plugin/pull/55

**Comment by smerle33** [comment]
Checked again here : https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fjenkins-infra-test-plugin/detail/PR-55/5/pipeline/ and it's working

File: vars/buildPlugin.groovy [hunk 2]
```diff
@@ -29,24 +29,39 @@ def call(Map params = [:]) {
...
+    } else {
+      switch(platform) {
+        case 'windows':
+          label = 'docker-windows'
+          break
+        case 'linux':
+          label = 'vm && linux'
+          break
+        default:
+          echo "WARNING: Unknown platform '${platform}'. Agent label set to fallback value 'linux'"
```
Comment (jglick) [inline vars/buildPlugin.groovy:62]: This is incorrect. The label should be left as is. For example https://github.com/jenkinsci/docker-workflow-plugin/runs/9836537148 ran the `maven-windows` branch from https://github.com/jenkinsci/docker-workflow-plugin/blob/d5d2e5c4007f7ea006152542b2bcbe0f1b2b08aa/Jenkinsfile#L3 on Linux (duplicating the designated Linux branch).

Comment (dduportal) [reply vars/buildPlugin.groovy:62]: https://github.com/jenkins-infra/pipeline-library/pull/658

**Pull request merged**